### PR TITLE
[XS][1648] Add noWrap to team name display in `TeamCharacterSelector`

### PR DESCRIPTION
**Pull request by BaruYogesh**
## Describe your changes

My first PR in this repo. Not familiar with mUI so bear with me. I just added a noWrap to a typography element, then added a nested typography element (is this ok?). i needed a nested element in order to keep using flex on the parent element, as ellipses wrap only works for inline display

willing to add more to other relevant areas when i hear that this is an acceptable pattern for approach.

## Issue or discord link

https://github.com/frzyc/genshin-optimizer/issues/1648

## Testing/validation

https://github.com/user-attachments/assets/522260cc-66c6-4bab-ac69-b6a25a37fb75

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [x] I have commented my code in hard-to understand areas.
- [x] I have made corresponding changes to README or wiki.
- [x] For front-end changes, I have updated the corresponding English translations.
- [x] I have run `yarn run mini-ci` locally to validate format and lint.
- [x] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

## Summary by CodeRabbit

- **New Features**
	- Enhanced styling and readability of the TeamCharacterSelector component.
	- Improved text wrapping for the team name display to prevent overflow.
	- Added a new Typography component to display the active build name alongside the character's name.
	- Updated dropdown button styling to handle long text with ellipsis.

- **Bug Fixes**
	- Adjusted properties to ensure the team name is displayed correctly without layout issues.

- **Refactor**
	- Streamlined the JSX structure for better clarity and maintainability.
	- Refined layout and styling of the LoadoutDropdown component for improved text handling.
	- Enhanced the BuildCard component's title styling for better presentation.

### CODE REVIEW

File: libs/gi/page-team/src/TeamCharacterSelector.tsx [hunk 1]
```diff
@@ -146,17 +152,19 @@ export default function TeamCharacterSelector({
             }}
           >
             <Typography
```
Comment (frzyc) [inline libs/gi/page-team/src/TeamCharacterSelector.tsx:154]: I think we can simplify the hierarchy:
-CardContent (flex, gap, align+justify:center, hover)
  - TeamIcon
  - Typography ( nowrap, h5, textshadow)

**Review by frzyc — CHANGES_REQUESTED** [review]
I think there is another location to apply this change, according to the issue:
https://github.com/frzyc/genshin-optimizer/blob/master/libs/gi/page-team/src/CharacterDisplay/Content.tsx#L168

File: libs/gi/page-team/src/TeamCharacterSelector.tsx (hunk 1 above)
Comment (BaruYogesh) [reply libs/gi/page-team/src/TeamCharacterSelector.tsx]: this worked 👍

**Comment by BaruYogesh** [comment]
> I think there is another location to apply this change, according to the issue: https://github.com/frzyc/genshin-optimizer/blob/master/libs/gi/page-team/src/CharacterDisplay/Content.tsx#L168

Screenshot 2024-10-24 at 10 09 14 AM

absolutely, i wanted to verify the approach before adding to other spots. let me know if this all looks right or if there are other spots you are thinking of (new to this app)

**Comment by frzyc** [comment]
> > I think there is another location to apply this change, according to the issue: https://github.com/frzyc/genshin-optimizer/blob/master/libs/gi/page-team/src/CharacterDisplay/Content.tsx#L168
> 
> Screenshot 2024-10-24 at 10 09 14 AM
> absolutely, i wanted to verify the approach before adding to other spots. let me know if this all looks right or if there are other spots you are thinking of (new to this app)

I wonder what happens when you have both a long loadout name and a long build name?

**Comment by frzyc** [comment]
> > > I think there is another location to apply this change, according to the issue: https://github.com/frzyc/genshin-optimizer/blob/master/libs/gi/page-team/src/CharacterDisplay/Content.tsx#L168
> > 
> > 
> > Screenshot 2024-10-24 at 10 09 14 AM
> > absolutely, i wanted to verify the approach before adding to other spots. let me know if this all looks right or if there are other spots you are thinking of (new to this app)
> 
> I wonder what happens when you have both a long loadout name and a long build name?
Looks like need to nowrap the build name as well
image

**Comment by BaruYogesh** [comment]
Screenshot 2024-10-26 at 6 35 16 PM
fixed! new to this app thanks for informing.

Screenshot 2024-10-26 at 6 35 10 PM
i tried to use typography nowrap for the buildcard, didnt quite turn out how i like lol

**Review by frzyc — APPROVED** [review]
LGTM. thanks for your contribution!

**Pull request merged**